### PR TITLE
DEV-2293: Allow GTM in CSP

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Allows Google Tag Manager in our CSP configuration.

#### Why are we doing this? How does it help us?

We use GTM for tracking, and this gets us closer to being able to turn on CSP enforcement.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Go to the donation thank-you page for Billy Penn in the review app.
2. Confirm that there is no CSP-related error in dev tools console, and the network tab in dev tools shows Google Tag Manager loading successfully.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

No, but I'd really like to. I'm not sure how to write a test for this kind of change, but it feels like it would be very valuable. I did some Googling around for ideas but didn't have much success.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

Lack of tests

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2293

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

I don't think so but I am not 100% sure.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: revengine/settings/base.py [hunk 1]
```diff
@@ -325,6 +325,7 @@
     "'self'",
     "https://js.stripe.com",
     "https://risk.clearbit.com",
+    "https://www.googletagmanager.com",
```
Comment (benjaminEwhite) [inline revengine/settings/base.py:328]: Wild!

**Review by benjaminEwhite — APPROVED** [review]
Code change is fine.

As for your comment on testing, i threw together one way we could test, but not sure how valuable it is.  Inside revengine.tests.test_views, we could add a test along these lines

```
from django.conf import settings


def test_csp_headers(client):

    setting_to_header_name = {
        "CSP_DEFAULT_SRC": "default-src",
        "CSP_SCRIPT_SRC": "script-src",
        "CSP_STYLE_SRC": "style-src",
        "CSP_IMG_SRC": "img-src",
        "CSP_FONT_SRC": "font-src",
        "CSP_FRAME_SRC": "frame-src",
        "CSP_CONNECT_SRC": "connect-src",
        "CSP_OBJECT_SRC": "object-src",
    }
    response = client.get("")
    csps = {}
    # CSP header splits rule for each content source by ";".
    # The first item per source is the source name (for instance, "style-src")
    # and then each item for the policy is split by space.
    for x in response.headers["Content-Security-Policy"].split(";"):
        split = [y.strip() for y in x.strip().split(" ")]
        csps[split[0]] = split[1:]
    for k, v in setting_to_header_name.items():
        assert set(getattr(settings, k)) == set(x for x in csps[v] if "nonce" not in x)
    for src in settings.CSP_INCLUDE_NONCE_IN:
        assert next(x for x in csps[src] if x.startswith("'nonce-"))
```

On one hand, this is kind of testing against 3rd party package (django-csp), in which case not super valuable.

On other hand, it's a guard against inadvertently changing the CSP settings in settings, which is possibly valuable.

@x110dc -- any thoughts on this approach to testing?

**Comment by nrh-cklimas** [comment]
@benjaminEwhite (and @x110dc) I don't have a firm idea on this, but the code snippet to get the generated policy seems useful. It would be nice if there was a magical Python library that emulated a browser and validated whether a request to a particular resource would be allowed or not, but I wasn't able to find much on PyPI. I don't think Cypress could be used for this (yet) because it runs directly against the SPA. My gut wants to suggest that this could be a snapshot test, e.g. we want to make sure the policy only changes when we expect it to but completely parsing the policy is unnecessary, but I don't think we have snapshot pytest tests set up.

**Review by x110dc — APPROVED** [review]
I agree that we shouldn't need to test much as this is more of a configuration change than a code change and the third party library presumably has its own tests. We could add the test Ben describes to guard against inadvertent changes to the config. Could also test for the presence of something like `https://js.stripe.com` in the CSP headers. 

I'm okay with it as-is, but okay with adding that test(s) if you'd like.

**Pull request merged**